client/firecracker_client_factory_test.go

```diff
@@ -0,0 +1,47 @@
+package client
+
+import (
+	"github.com/stretchr/testify/assert"
+	"net"
+	"testing"
+	"time"
+)
+
+func TestNewHttpClientForUNIXSocket(t *testing.T) {
+	done := make(chan bool)
+
+	expectedMessage := "PUT /logger HTTP/1.1\r\nHost: localhost\r\nUser-Agent: Go-http-client/1.1\r\nContent-Length: 0\r\nAccept: application/json\r\nContent-Type: application/json\r\nAccept-Encoding: gzip\r\n\r\n"
+	socketName := "testingUnixSocket.sock"
+	addr, _ := net.ResolveUnixAddr("unix", socketName)
+	listener, _ := net.ListenUnix("unix", addr)
+	defer listener.Close()
+
+	go func() {
+		conn, err := listener.AcceptUnix()
```

Comment:
Unix file needs to be closed and cleaned up.

Reply:
If I'm not mistaken, defer is closing the unix file.  
I was able to verify that the socket file is being deleted after test execution